src/cmap/auth/mongodb_oidc.ts

```diff
@@ -11,29 +17,91 @@ export interface OIDCMechanismServerStep1 {
   requestScopes?: string[];
 }
 
-/**
- * TODO: NODE-5035: Make API public
- *
- * @internal */
+/** @public */
 export interface OIDCRequestTokenResult {
   accessToken: string;
   expiresInSeconds?: number;
   refreshToken?: string;
 }
 
-/**
- * TODO: NODE-5035: Make API public
- *
- * @internal */
+/** @public */
 export type OIDCRequestFunction = (
-  idl: OIDCMechanismServerStep1
+  principalName: string,
+  serverResult: OIDCMechanismServerStep1,
+  timeout: AbortSignal | number
 ) => Promise<OIDCRequestTokenResult>;
 
-/**
- * TODO: NODE-5035: Make API public
- *
- * @internal */
+/** @public */
 export type OIDCRefreshFunction = (
-  idl: OIDCMechanismServerStep1,
-  result: OIDCRequestTokenResult
+  principalName: string,
+  serverResult: OIDCMechanismServerStep1,
+  result: OIDCRequestTokenResult,
+  timeout: AbortSignal | number
 ) => Promise<OIDCRequestTokenResult>;
+
+/** @internal */
+export const OIDC_WORKFLOWS = {
+  callback: new CallbackWorkflow(),
+  aws: new AwsDeviceWorkflow(),
+  azure: undefined,
+  gcp: undefined
```

Comment:
```suggestion
  gcp: undefined,
  __proto__: null
```

since otherwise the `OIDC_WORKFLOWS[deviceName || 'callback']` below would return something for e.g. `deviceName === 'hasOwnProperty'`… :)

Reply:
Just some ideas, we aren't using this object as an enum per se, it serves as a lookup, so the losing the `as const`-ness of it wouldn't harm the usability

```typescript
export const OIDC_WORKFLOWS = Object.create(null) as { [deviceName: string]: Workflow | undefined };
OIDC_WORKFLOWS.callback = new CallbackWorkflow();
OIDC_WORKFLOWS.aws = new AwsDeviceWorkflow();
```

Alternatively, making a Map, and relying on `.get` returning undefined / defaulting the key to `'callback'` also serves us here.

```typescript
export const OIDC_WORKFLOWS = new Map<string, Workflow>([
  ['callback', new CallbackWorkflow()],
  ['aws', new AwsDeviceWorkflow()],
])
```

Would we prefer one of these to the proto setter?

Reply:
> Would we prefer one of these to the proto setter?

So, no strong feelings here, but maybe to clarify: `__proto__` inside object literals is special syntax and entirely distinct from the legacy `Object.prototype.__proto__` getter/setter (other than using the same identifier).

(Fun JS trivia fact: `eval('({ "__proto__": null })')` and `JSON.parse('{ "__proto__": null }')` are different things…)

I’d say it’s fairly standard to go with `{ ..., __proto__: null }` for lookup objects, especially when it’s a constant list and you don’t need all the flexibility that a map gives you.

Reply:
the _proto_ that keeps on giving! TIL, thanks for the knowledge share, also no strong feelings, this gets the job done. @durran just offering here that, now that we have Maps we don't need to create lookup objects with a special consideration for inherited keys, if we want to take that approach